cdap-api/src/main/java/io/cdap/cdap/api/app/RuntimeConfigurer.java

```diff
@@ -37,5 +37,6 @@ public interface RuntimeConfigurer extends ServiceDiscoverer {
    * for preview run, there is no existing app spec
    */
   @Nullable
+  @Deprecated
```

Comment:
please add/update javadoc to recommend using  ApplicationConfigurer#getDeployedApplicationSpec